src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/CustomConverterTests/CustomConverterTests.KeyConverters.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Encodings.Web;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class CustomConverterTests
+    {
+        [Theory]
+        [MemberData(nameof(GetCustomKeysAndExpectedPropertyNameEncodings))]
+        public static void CustomKeyConverter_Serialization<TKey>(TKey key, string expectedKeyEncoding)
+        {
+            var options = new JsonSerializerOptions { Converters = { new EmbeddedJsonKeyConverter<TKey>() } };
```

Comment:
style: I've been leaning towards this syntax:
```cs
JsonSerializerOptions options = new() { Converters = { new EmbeddedJsonKeyConverter<TKey>() } };
```
since I find the Type name upfront is a bit easier to read...